src/Avalonia.Controls/GridSplitter.cs

```diff
@@ -205,13 +206,50 @@ private void RemovePreviewAdorner()
             }
         }
 
+        private Grid? GetParentGrid()
+        {
+            var grid = Parent as Grid;
+            if (grid is not null)
+            {
+                return grid;
+            }
+
+            // When GridSplitter is used inside an ItemsControl with Grid as
+            // its ItemsPanel, its immediate parent is usually a ItemsControl or ContentPresenter.
+            switch (Parent)
+            {
+                case ItemsControl itemsControl:
+                {
+                    if (itemsControl.ItemsPanelRoot is Grid itemsPanelGrid)
+                    {
+                        grid = itemsPanelGrid;
+                    }
+
+                    break;
+                }
+                case ContentPresenter { Parent: ItemsControl presenterItemsControl }:
```

Comment:
I'm not sure I like having such a specialization here. It seems a bit odd to have `GridSplitter` know about every possible hierarchy it could be in.

That said, introducing some abstraction just for this purpose sounds like over engineering...

Reply:
Exposed the helper methods as protected virtual https://github.com/AvaloniaUI/Avalonia/pull/19200/commits/e1fdb328730f118c8e7c71fbb90b3d4e1ec4a241 might come useful is custom control scenarios